server/src/thirdparty/slackAPI/controller.js

```diff
@@ -0,0 +1,82 @@
+import { sendSuccessMessage, sendFailureMessage } from '../utils/restUtil';
+import slackService from './service';
+import config from './config';
+
+const authCallBack = async (req, res) => {
+  try {
+    const { code, state } = req.query;
+    if (code === undefined || code === '') {
+      sendFailureMessage(res, 'Code is empty', 400);
+    } else if (state === undefined || state === '') {
+      sendFailureMessage(res, 'User Token is empty', 400);
+    } else {
+      const authData = await slackService.generateAuthToken(code);
+      const status = await slackService.addOrUpdateIntegration(authData, state);
+      if (status) {
+        sendSuccessMessage(res, 'Authorization success');
+      } else {
+        sendFailureMessage(res, 'Authorization failed');
+      }
+    }
+  } catch (err) {
+    sendFailureMessage(res, err.message);
+  }
+};
+
+const eventCallBack = async (req, res) => {
+  try {
+    // eslint-disable-next-line camelcase
+    const { team_id, authed_users, event } = req.body;
+    const eventKey = `${event.type}_${event.channel_type}`;
+    if (config.event[eventKey] !== undefined) {
+      slackService.generateNewEvent(team_id, authed_users, event);
+    }
+    sendSuccessMessage(res, 'Event received');
+  } catch (err) {
+    sendFailureMessage(res, err.message);
+  }
+};
+
+const integrateSlack = async (req, res) => {
+  const { code } = req.params;
+  if (code === undefined || code === '') {
+    sendFailureMessage(res, 'Code should not be empty', 400);
+  } else {
+    const integUrl = slackService.getIntegUrl(code);
+    res.redirect(integUrl);
+  }
+};
+
+const getEvents = (_, res) => {
+  sendSuccessMessage(res, config.event);
+};
+
+const getActions = (_, res) => {
+  sendSuccessMessage(res, config.action);
+};
+
+const getAccountData = async (req, res) => {
```

Comment:
async without await is not a good idea. 

`slackService.getAccountData` can be put in awiat and try/catch. Please refactor

Reply:
Done